reactnative/RNPlayground/app/screens/PlaygroundScreen.tsx

```diff
@@ -1,108 +1,96 @@
-    }, [playgroundStore])
-  )
-
-  return <View style={$container}>
+export const PlaygroundScreen: React.FC<PlaygroundScreenProps> = observer(({ navigation }) => {
+  const items: PlaygroundCardProps[] = [
+    photosItem(navigation),
+    clickMeItem,
     {
-      StateComponent(
-        $bottomContainerInsets,
-        playgroundStore.uiState,
-        () => playgroundStore.resetCounter()
-      )
-    }
-  </View>
-})
+      title: "I do nothing",
+      color: "forestgreen",
+      onPress: () => {
+      },
+    },
+  ]
 
-function StateComponent(insets, uiState, onResetClicked) {
-  const { loading, error, content } = uiState
-  if (loading) {
-    return LoadingComponent(insets)
-  } else if (error) {
-    return ErrorComponent(insets, error, onResetClicked)
-  } else if (content) {
-    return ContentComponent(insets, content, onResetClicked)
-  }
-  throw new Error(`UiState type is not handled ${JSON.stringify(uiState)}`)
-}
-
-function ContentComponent(insets, content, onResetClicked) {
-  return <View>
-    <View style={$topContainer}>
-      <Text
-        testID="welcome-heading"
-        style={$welcomeHeading}
-        tx="playgroundScreen.readyForLaunch"
-        preset="heading"
+  return <FlatList
+    style={{
+      backgroundColor: colors.background,
+      flex: 1,
+    }}
+    data={items}
+    renderItem={({ item }) =>
+      <PlaygroundCard
+        title={item.title}
+        color={item.color}
+        onPress={item.onPress}
       />
-      <Text tx="playgroundScreen.postscript" preset="subheading" />
-    </View>
-
-    <View style={[$bottomContainer, insets]}>
-      <Text tx="playgroundScreen.exciting" size="md" />
-      <Text text={`Value is: ${content.number}`} size="md" />
-      <Button text="Reset" onPress={onResetClicked} />
-    </View>
-  </View>
-}
-
-function ErrorComponent(insets, error, onResetClicked) {
-  return <View style={[insets]}>
-    <Text text={`ERROR: ${error.message}`} size="md" />
-    <Button text="Refresh" onPress={onResetClicked} />
-  </View>
-}
-
-function LoadingComponent(insets) {
-  return <View style={[insets, { justifyContent: "space-around", flexGrow: 1 }]}>
-    <Text text={`LOADING`} size="md" style={{ textAlign: "center" }} />
-  </View>
-}
+    }
+  >
+  </FlatList>
+})
 
-const $container: ViewStyle = {
-  flex: 1,
-  backgroundColor: colors.background,
-}
+const photosItem = (navigation) => ({
+  title: translate("playgroundScreen.photos"),
+  onPress: () => navigation.navigate("Photos"),
+  color: colors.palette.accent200,
+})
 
-const $topContainer: ViewStyle = {
-  flexShrink: 1,
-  flexGrow: 1,
-  flexBasis: "57%",
-  justifyContent: "center",
-  paddingHorizontal: spacing.large,
+const clickMeItem = {
+  title: "Click me",
+  color: colors.palette.accent300,
+  onPress: () => {
+    if (Platform.OS == "android") {
+      ToastAndroid.show("You're Android, here's a toast to you", ToastAndroid.SHORT)
+    } else {
+      Alert.alert(
```

Comment:
This `Alert.alert()` is already a multuiplatform solution, buts its interesting to see how the device platform is checked with ease, and impls for native components

> This is an API that works both on iOS and Android and can show static alerts.

Reply:
I wanted to show a toast, but there's no toast on iOS which gave me a chance to see how to differentiate between platforms